File: frontend/site/src/app/model/Film.ts
import {FilmWIkiData} from "./FilmWIkiData";


export class Film {
  private user_id: number;
  private id: number;
  private created: Date;
  private version: Date;
  private codWikiData: string;
  private esteFavorit: boolean;
  private statusFilm: string;
  private notaFilm: number;
  private detaliiFilm: FilmWIkiData;

  constructor(user_id: number, id: number, created: Date, version: Date, codWikiData: string, esteFavorit: boolean, statusFilm: string, notaFilm: number, detaliiFilm: FilmWIkiData) {
    this.user_id = user_id;
    this.id = id;
    this.created = created;
    this.version = version;
    this.codWikiData = codWikiData;
    this.esteFavorit = esteFavorit;
    this.statusFilm = statusFilm;
    this.notaFilm = notaFilm;
    this.detaliiFilm = detaliiFilm;
  }
}

File: backend/src/main/java/com/bdsa/disertatie/backend/entity/Film.java
package com.bdsa.disertatie.backend.entity;


import com.bdsa.disertatie.backend.enums.StatusFilmEnum;

import javax.persistence.*;

@Entity
@Table(name = "film",
       uniqueConstraints = {@UniqueConstraint(name = "unique_cod_wiki_data",
                            columnNames = {"user_id","codWikiData"})})
public class Film extends BaseEntity{

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(nullable = false, name = "user_id")
    private Utilizator utilizator;
    @Column(nullable = false)
    private String codWikiData;
    @Column(columnDefinition="BOOLEAN DEFAULT false", nullable = false)
    private Boolean esteFavorit;
    @Enumerated(EnumType.STRING) //pentru a salva valoarea textuala din enum (nu codul)
    private StatusFilmEnum statusFilm;
    private Integer notaFilm;

    public Utilizator getUtilizator() {
        return utilizator;
    }

    public void setUtilizator(Utilizator utilizator) {
        this.utilizator = utilizator;
    }

    public String getCodWikiData() {
        return codWikiData;
    }

    public void setCodWikiData(String codWikiData) {
        this.codWikiData = codWikiData;
    }

    public Boolean getEsteFavorit() {
        return esteFavorit;
    }

    public void setEsteFavorit(Boolean esteFavorit) {
        this.esteFavorit = esteFavorit;
    }

    public StatusFilmEnum getStatusFilm() {
        return statusFilm;
    }

    public void setStatusFilm(StatusFilmEnum statusFilm) {
        this.statusFilm = statusFilm;
    }

    public Integer getNotaFilm() {
        return notaFilm;
    }

    public void setNotaFilm(Integer notaFilm) {
        this.notaFilm = notaFilm;
    }
}
